src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapSessionOptions.Linux.cs

```diff
@@ -9,11 +9,12 @@ public partial class LdapSessionOptions
     {
         private static void PALCertFreeCRLContext(IntPtr certPtr) { /* No op */ }
 
-        [SupportedOSPlatform("windows")]
-        public bool SecureSocketLayer
+        public bool SecureSocketLayer { get; set; }
+
+        public int ProtocolVersion
         {
-            get => throw new PlatformNotSupportedException();
-            set => throw new PlatformNotSupportedException();
+            get => GetPtrValueHelper(LdapOption.LDAP_OPT_VERSION).ToInt32();
```

Comment:
> Fixes a problem with setting the LDAP version on the connection by using IntPtr types rather than int directly.

Could you say more?

Reply:
In my testing, I had a problem where the version was not being set properly when using an integer rather than an IntPtr, so it was defaulting to LDAPv2 on my machine and failing to connect. The API calls for an pointer to an int:

> LDAP_OPT_PROTOCOL_VERSION
>              Sets/gets the protocol version.  outvalue and  invalue  must  be
>              int *.

I think this might work now because the value is being [cast to an integer pointer on the C side](https://git.openldap.org/openldap/openldap/-/blob/OPENLDAP_REL_ENG_2_4_58/libraries/libldap/options.c#L851-852), but from what I understand (which is very little) I think that this can fail on systems with different pointer sizes?